Problem compiling ESP32_ESP-IDF components

Hello,
Maybe not an issue as such, maybe just me doing it wrong?
I'm trying to compile the ESP32_ESP-IDF components with IDF V3.3 but get a few errors:

r@r-PC MINGW32 ~/esp/esp-idf/ESP32_ESP-IDF
$ make -j9 flash
Toolchain path: /opt/xtensa-esp32-elf/bin/xtensa-esp32-elf-gcc
Toolchain version: crosstool-ng-1.22.0-80-g6c4433a5
Compiler version: 5.2.0
CXX build/I2Cdev/I2Cdev.o
CXX build/MPU6050/MPU6050.o
CXX build/cxx/cxx_exception_stubs.o
C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.cpp: In static member function 'static int8_t I2Cdev::readBytes(uint8_t, uint8_t, uint8_t, uint8_t*, uint16_t)':
C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.cpp:138:57: error: invalid conversion from 'int' to 'i2c_ack_type_t' [-fpermissive]
   ESP_ERROR_CHECK(i2c_master_read(cmd, data, length-1, 0));
                                                         ^
C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.cpp:45:51: note: in definition of macro 'ESP_ERROR_CHECK'
 #define ESP_ERROR_CHECK(x)   do { esp_err_t rc = (x); if (rc != ESP_OK) { ESP_LOGE("err", "esp_err_t = %d", rc); /*assert(0 && #x);*/} } while(0);
                                                   ^
In file included from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.h:37:0,
                 from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.cpp:40:
C:/msys32/home/r/esp/esp-idf/components/driver/include/driver/i2c.h:325:11: note:   initializing argument 4 of 'esp_err_t i2c_master_read(i2c_cmd_handle_t, uint8_t*, size_t, i2c_ack_type_t)'
 esp_err_t i2c_master_read(i2c_cmd_handle_t cmd_handle, uint8_t* data, size_t data_len, i2c_ack_type_t ack);
           ^
C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.cpp:140:60: error: invalid conversion from 'int' to 'i2c_ack_type_t' [-fpermissive]
  ESP_ERROR_CHECK(i2c_master_read_byte(cmd, data+length-1, 1));
                                                            ^
C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.cpp:45:51: note: in definition of macro 'ESP_ERROR_CHECK'
 #define ESP_ERROR_CHECK(x)   do { esp_err_t rc = (x); if (rc != ESP_OK) { ESP_LOGE("err", "esp_err_t = %d", rc); /*assert(0 && #x);*/} } while(0);
                                                   ^
In file included from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.h:37:0,
                 from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.cpp:40:
C:/msys32/home/r/esp/esp-idf/components/driver/include/driver/i2c.h:306:11: note:   initializing argument 3 of 'esp_err_t i2c_master_read_byte(i2c_cmd_handle_t, uint8_t*, i2c_ack_type_t)'
 esp_err_t i2c_master_read_byte(i2c_cmd_handle_t cmd_handle, uint8_t* data, i2c_ack_type_t ack);
           ^
C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.cpp: In static member function 'static bool I2Cdev::writeWord(uint8_t, uint8_t, uint16_t)':
C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.cpp:152:10: warning: unused variable 'data2' [-Wunused-variable]
  uint8_t data2[] = {(uint8_t)(data & 0xff), (uint8_t)(data>>8)};
          ^
CC build/console/linenoise/linenoise.o
CC build/coap/libcoap/src/address.o
In file included from C:/msys32/home/r/esp/esp-idf/components/driver/include/driver/i2c.h:23:0,
                 from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.h:37,
                 from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/MPU6050/MPU6050.h:41,
                 from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/MPU6050/MPU6050.cpp:37:
C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/MPU6050/MPU6050.cpp: In member function 'void MPU6050::ReadRegister(uint8_t, uint8_t*, uint8_t)':
C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/MPU6050/MPU6050.cpp:52:52: error: invalid conversion from 'int' to 'i2c_ack_type_t' [-fpermissive]
   ESP_ERROR_CHECK(i2c_master_read(cmd, data, len, 0));
                                                    ^
C:/msys32/home/r/esp/esp-idf/components/esp32/include/esp_err.h:117:31: note: in definition of macro 'ESP_ERROR_CHECK'
         esp_err_t __err_rc = (x);                                       \
                               ^
In file included from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.h:37:0,
                 from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/MPU6050/MPU6050.h:41,
                 from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/MPU6050/MPU6050.cpp:37:
C:/msys32/home/r/esp/esp-idf/components/driver/include/driver/i2c.h:325:11: note:   initializing argument 4 of 'esp_err_t i2c_master_read(i2c_cmd_handle_t, uint8_t*, size_t, i2c_ack_type_t)'
 esp_err_t i2c_master_read(i2c_cmd_handle_t cmd_handle, uint8_t* data, size_t data_len, i2c_ack_type_t ack);
           ^
In file included from C:/msys32/home/r/esp/esp-idf/components/driver/include/driver/i2c.h:23:0,
                 from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.h:37,
                 from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/MPU6050/MPU6050.h:41,
                 from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/MPU6050/MPU6050.cpp:37:
C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/MPU6050/MPU6050.cpp:54:57: error: invalid conversion from 'int' to 'i2c_ack_type_t' [-fpermissive]
  ESP_ERROR_CHECK(i2c_master_read_byte(cmd, data+len-1, 1));
                                                         ^
C:/msys32/home/r/esp/esp-idf/components/esp32/include/esp_err.h:117:31: note: in definition of macro 'ESP_ERROR_CHECK'
         esp_err_t __err_rc = (x);                                       \
                               ^
In file included from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.h:37:0,
                 from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/MPU6050/MPU6050.h:41,
                 from C:/msys32/home/r/esp/esp-idf/ESP32_ESP-IDF/components/MPU6050/MPU6050.cpp:37:
C:/msys32/home/r/esp/esp-idf/components/driver/include/driver/i2c.h:306:11: note:   initializing argument 3 of 'esp_err_t i2c_master_read_byte(i2c_cmd_handle_t, uint8_t*, i2c_ack_type_t)'
 esp_err_t i2c_master_read_byte(i2c_cmd_handle_t cmd_handle, uint8_t* data, i2c_ack_type_t ack);
           ^
make[1]: *** [/home/r/esp/esp-idf/make/component_wrapper.mk:290: I2Cdev.o] Error 1
make: *** [C:/msys32/home/r/esp/esp-idf/make/project.mk:552: component-I2Cdev-build] Error 2
make: *** Waiting for unfinished jobs....
CC build/coap/libcoap/src/async.o
make[1]: *** [/home/r/esp/esp-idf/make/component_wrapper.mk:290: MPU6050.o] Error 1
make: *** [C:/msys32/home/r/esp/esp-idf/make/project.mk:552: component-MPU6050-build] Error 2

Can anyone please suggest where I'm going wrong?
Thanks in advance

--- a/ESP32_ESP-IDF/components/I2Cdev/CMakeLists.txt
+++ b/ESP32_ESP-IDF/components/I2Cdev/CMakeLists.txt
@@ -1,0 +1,3 @@
+idf_component_register(SRCS "I2Cdev.cpp"
+                       INCLUDE_DIRS "."
+)

--- a/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.cpp
+++ b/ESP32_ESP-IDF/components/I2Cdev/I2Cdev.cpp
@@ -135,9 +135,9 @@ int8_t I2Cdev::readBytes(uint8_t devAddr, uint8_t regAddr, uint8_t length, uint8
 	ESP_ERROR_CHECK(i2c_master_write_byte(cmd, (devAddr << 1) | I2C_MASTER_READ, 1));
 
 	if(length>1)
-		ESP_ERROR_CHECK(i2c_master_read(cmd, data, length-1, 0));
+		ESP_ERROR_CHECK(i2c_master_read(cmd, data, length-1, I2C_MASTER_ACK));
 
-	ESP_ERROR_CHECK(i2c_master_read_byte(cmd, data+length-1, 1));
+	ESP_ERROR_CHECK(i2c_master_read_byte(cmd, data+length-1, I2C_MASTER_NACK));
 
 	ESP_ERROR_CHECK(i2c_master_stop(cmd));
 	ESP_ERROR_CHECK(i2c_master_cmd_begin(I2C_NUM, cmd, 1000/portTICK_PERIOD_MS));

--- a/ESP32_ESP-IDF/components/MPU6050/CMakeLists.txt
+++ b/ESP32_ESP-IDF/components/MPU6050/CMakeLists.txt
@@ -1,0 +1,4 @@
+idf_component_register(SRCS "MPU6050.cpp"
+                       INCLUDE_DIRS "."
+                       REQUIRES I2Cdev
+)

--- a/ESP32_ESP-IDF/components/MPU6050/MPU6050.cpp
+++ b/ESP32_ESP-IDF/components/MPU6050/MPU6050.cpp
@@ -49,9 +49,9 @@ void MPU6050::ReadRegister(uint8_t reg, uint8_t *data, uint8_t len){
 	ESP_ERROR_CHECK(i2c_master_write_byte(cmd, (dev << 1) | I2C_MASTER_READ, 1));
 
 	if(len>1)
-		ESP_ERROR_CHECK(i2c_master_read(cmd, data, len, 0));
+		ESP_ERROR_CHECK(i2c_master_read(cmd, data, len, I2C_MASTER_ACK));
 
-	ESP_ERROR_CHECK(i2c_master_read_byte(cmd, data+len-1, 1));
+	ESP_ERROR_CHECK(i2c_master_read_byte(cmd, data+len-1, I2C_MASTER_NACK));
 
 	ESP_ERROR_CHECK(i2c_master_stop(cmd));
 	ESP_ERROR_CHECK(i2c_master_cmd_begin(I2C_NUM, cmd, 1000));

